pinot-transport/src/test/java/org/apache/pinot/transport/common/ServerInstanceTest.java

```diff
@@ -18,61 +18,27 @@
 import org.apache.pinot.common.response.ServerInstance;
-import org.testng.Assert;
 import org.testng.annotations.Test;
 
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotEquals;
+
 
 public class ServerInstanceTest {
 
   @Test
-  public void testServerInstance()
-      throws Exception {
-    // Same local hostname and port
-    {
-      ServerInstance instance1 = new ServerInstance("localhost", 8080);
-      ServerInstance instance2 = new ServerInstance("localhost", 8080);
-//      System.out.println("Instance 1 :" + instance1);
-//      System.out.println("Instance 2 :" + instance2);
-      Assert.assertEquals(instance2, instance1, "Localhost server-instances with same port");
-    }
-
-    // Same  hostname and port
-    {
-      ServerInstance instance1 = new ServerInstance("test-host", 8080);
-      ServerInstance instance2 = new ServerInstance("test-host", 8080);
-//      System.out.println("Instance 1 :" + instance1);
-//      System.out.println("Instance 2 :" + instance2);
-      Assert.assertEquals(instance2, instance1, "Localhost server-instances with same port");
-    }
-    // same hostname but different port
-    {
-      ServerInstance instance1 = new ServerInstance("localhost", 8081);
-      ServerInstance instance2 = new ServerInstance("localhost", 8082);
-//      System.out.println("Instance 1 :" + instance1);
-//      System.out.println("Instance 2 :" + instance2);
-      Assert.assertFalse(instance1.equals(instance2), "Localhost server-instances with same port");
-    }
+  public void testServerInstance() {
+    // Same local host and port
+    assertEquals(new ServerInstance("localhost", 8080), new ServerInstance("localhost", 8080));
 
-    // same port but different host
-    {
-      ServerInstance instance1 = new ServerInstance("abcd", 8080);
-      ServerInstance instance2 = new ServerInstance("abce", 8080);
-//      System.out.println("Instance 1 :" + instance1);
-//      System.out.println("Instance 2 :" + instance2);
-      Assert.assertFalse(instance1.equals(instance2), "Localhost server-instances with same port");
-    }
+    // Same host and port
+    assertEquals(new ServerInstance("test-host", 8080), new ServerInstance("test-host", 8080));
 
-    // Test getIpAddress
-    {
-      InetAddress ipAddr = InetAddress.getByName("127.0.0.1");
+    // Same host but different port
+    assertNotEquals(new ServerInstance("localhost", 8081), new ServerInstance("localhost", 8082));
 
-      ServerInstance instance1 = new ServerInstance("127.0.0.1", 8080);
-      Assert.assertEquals(instance1.getPort(), 8080, "Port check");
-      Assert.assertEquals(instance1.getPort(), 8080, "Hostname check");
-      Assert.assertEquals(instance1.getHostname(), ipAddr.getHostName(), "Host check");
-      Assert.assertEquals(instance1.getIpAddress(), ipAddr, "IP check");
-    }
+    // Same port but different host
+    assertNotEquals(new ServerInstance("foo", 8080), new ServerInstance("bar", 8080));
```

Comment:
Can we include a test to test the case when host name is an ip address, like `127.0.0.1`?

Reply:
Done

Reply:
Thanks. How about the test comparing with `127.0.0.1` as hostname and `localhost` as hostname? Would they be the same? Can we include this test as well?

Reply:
That won't be the same as we do not resolve the ip address

Reply:
Added back the IP address lookup and added test